src/search/merge_and_shrink/transition_system.h

```diff
@@ -41,42 +39,43 @@ struct Transition {
     }
 };
 
-struct GroupAndTransitions {
-    const LabelGroup &label_group;
-    const std::vector<Transition> &transitions;
-    GroupAndTransitions(const LabelGroup &label_group,
-                        const std::vector<Transition> &transitions)
+using LabelGroup = std::vector<int>;
+
+class LocalLabelInfo {
+    LabelGroup label_group;
+    std::vector<Transition> transitions;
+    int cost;
+
+    friend class TransitionSystem;
```

Comment:
`friend` is generally a pretty bad code smell. Is it necessary?

Reply:
It allows `TransitionSystem` to modify the private members `label_group` , `transitions`, and `cost`. The alternative would be to add three non-const getters in addition to the existing const getters. Do you prefer that? Mir ist's egal.

Reply:
OK, I have to check how the class is used in general to understand better what the benefits/drawbacks of this would be. Give some time to clone the code and look through this.

Reply:
It looks like the only non-const non-friend use of `LocalLabelInfo` is in `fts_factory.cc`. In `TransitionSystem` I also only found one non-const use, which changes three things at the same time. Would it make sense to update this change in a (non-const) method of LocalLabelInfo? Then we wouldn't need friend any more. With friend, we cannot be sure of the class invariants of `LocalLabelInfo` without checking all of `TransitionSystem`, which is a very large class and only messes with the internal details of `LocalLabelInfo` in a single place.

Reply:
I pushed a commit to remove the friend declaration. It adds several methods to LocalLabelInfo. Do you prefer this version over the alternative of turning LocalLabelInfo into a struct? For me, either way is fine.